lib/src/install.rs

```diff
@@ -1298,18 +1304,64 @@ async fn install_with_sysroot(
     tracing::debug!("Installed bootloader");
 
     tracing::debug!("Perfoming post-deployment operations");
-    // Note that we *always* initialize this container storage, even
-    // if there are no bound images today.
+
+    // Note that we *always* initialize this container storage, even if there are no bound images
+    // today.
     let imgstore = sysroot.get_ensure_imgstore()?;
-    // Now copy each bound image from the host's container storage into the target.
-    for image in bound_images {
-        let image = image.image.as_str();
-        imgstore.pull_from_host_storage(image).await?;
+
+    match bound_images {
+        BoundImages::Skip => {}
+        BoundImages::Resolved(resolved_bound_images) => {
+            // Now copy each bound image from the host's container storage into the target.
+            for image in resolved_bound_images {
+                let image = image.image.as_str();
+                imgstore.pull_from_host_storage(image).await?;
+            }
+        }
+        BoundImages::Unresolved(bound_images) => {
+            crate::boundimage::pull_images(sysroot, bound_images)
+                .await
+                .context("pulling bound images")?;
+        }
     }
 
     Ok(())
 }
 
+enum BoundImages {
+    Skip,
+    Resolved(Vec<ResolvedBoundImage>),
+    Unresolved(Vec<BoundImage>),
+}
+
+impl BoundImages {
+    async fn from_state(state: &State) -> Result<Self> {
+        let bound_images = (!state.config_opts.skip_bound_images)
+            .then(|| crate::boundimage::query_bound_images(&state.container_root))
+            .transpose()?;
+
+        Ok(match bound_images {
```

Comment:
Super minor style thing but I tend to prefer e.g.

```
let r = match { ... };
Ok(r)
```

To avoid the rightward-drift of the `Ok`, *or* moving the `Ok` into the match.

Reply:
>moving the Ok into the match.

Definitely not, followed extracted variable as I don't mind either way. But I'm curious, what do you mean by rightward-drift of the `Ok` here?

Reply:
Adding the `Ok(` shifts everything inside it by 3 columns

Reply:
Only the small part in the line with the `Ok(` itself, the rest of the lines of the match statement stay the same

Reply:
Yeah fair, marking resolved